### PR TITLE
New version of the cling patches

### PR DESCRIPTION
**Pull request by SylvainCorlay**
Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by SylvainCorlay** [comment]
cc @chrisburr  this works locally (except for the startwith issue on config values)

I wish the root and cling folks used a single set of patches, but unfortunately, there is no common set that works.

**Comment by chrisburr** [comment]
I'll take a look locally at the `startswith` issue. (Btw you'll need a rerender to pick up the changes to the build config.)

> I wish the root and cling folks used a single set of patches, but unfortunately, there is no common set that works.

Me too...I'm trying to push for it when they update from LLVM 5 but we'll see (which is increasingly urgent due to issues GCC 9).

**Comment by SylvainCorlay** [comment]
@chrisburr tests appear to be passing.

Leving this open for a bit to give you a chance to comment!

**Comment by isuruf** [comment]
Are these variant clangdev packages used as static libraries or as tools? I'm wondering whether there's any benefit to having variants

**Comment by chrisburr** [comment]
They're used as static libraries

File: recipe/meta.yaml [hunk 1]
```diff
@@ -12,16 +16,19 @@ source:
...
+  git_rev: {{ clang_variant|replace('root_', '')|replace('.', '-') }}
   folder: root-source
 {% endif %}
 
 build:
   number: {{ build_number }}
-  skip: true  # [win and vc<14]
-
+  # skip: true  # [win and vc<14]
+  skip: true  # [win]
```
Comment (chrisburr) [inline recipe/meta.yaml]: Was this intentional? I guess it was done for the rerendering bug? If so, I previously worked around it by removing this line, deleting the vc9 build config then re-adding this line.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -12,16 +16,19 @@ source:
   fn: cfe-{{ version }}.src.tar.xz
```
Comment (chrisburr) [inline recipe/meta.yaml]: This line can be dropped
```suggestion
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,7 +1,11 @@
 {% set version = "5.0.0" %}
 {% set sha256 = "019f23c2192df793ac746595e94a403908749f8e0c484b403476d2611dd20970" %}
 
-{% set build_number = "1005" %}
+{% set build_number = "1006" %}
```
Comment (chrisburr) [inline recipe/meta.yaml]: Is there any need to bump the build number? I guess there is no need to redeploy the other variants.
```suggestion
{% set build_number = "1005" %}
```

Comment (SylvainCorlay) [reply recipe/meta.yaml]: I guess that this is fine to leave it at 1005.

File: recipe/meta.yaml (hunk 1 above)
Comment (SylvainCorlay) [reply recipe/meta.yaml]: ok, will try that.

**Comment by isuruf** [comment]
If they are used as static libraries, why create variants? ROOT can build this in their build system right?

**Comment by SylvainCorlay** [comment]
> If they are used as static libraries, why create variants? ROOT can build this in their build system right?

Cling also uses such a variant. It seems to me that it would complicate the build quite a bit to have to build clang completely as part of cling.

**Comment by SylvainCorlay** [comment]
now the missing libarchive bug.

**Pull request merged**